clippy_lints/src/loops/manual_flatten.rs

```diff
@@ -28,7 +28,7 @@ pub(super) fn check<'tcx>(
         && let PatKind::Binding(_, pat_hir_id, _, _) = pat.kind
         && path_to_local_id(let_expr, pat_hir_id)
         // Ensure the `if let` statement is for the `Some` variant of `Option` or the `Ok` variant of `Result`
-        && let PatKind::TupleStruct(ref qpath, _, _) = let_pat.kind
+        && let PatKind::TupleStruct(ref qpath, pats, _) = let_pat.kind
```

Comment:
We can destructure here rather than iterating

```suggestion
        && let PatKind::TupleStruct(ref qpath, [inner_pat], _) = let_pat.kind
```